Automattic/woocommerce-payments
# Track platform checkout OTP attempts

Fixes #3830 

#### Changes proposed in this Pull Request

Implements the following Tracks events to record the platform-checkout OTP usage
* `woocommerceanalytics_platform_checkout_otp_prompt_start`
* `woocommerceanalytics_platform_checkout_otp_prompt_complete`
* `woocommerceanalytics_platform_checkout_otp_prompt_failed`

See pdpOw2-l8-p2, pdpOw2-mz-p2

This PR depends on the changes in #506 in platform-checkout repository










#### Testing instructions





**Setup**

* Ensure the changes in PR #506 in platform-checkout repository is available locally.
* In your `wp_options` table.
  * Ensure `jetpack_tos_agreed` is set to 1. This will already be set if you have Jetpack plugin installed.
  * `_wcpay_feature_platform_checkout` option is set to 1.
* Enable Platform Checkout option as a merchant from Payments -> Settings -> Express checkouts.
* Add a product to the cart
* Visit blocks checkout
* Enter an email address of an existing platform-checkout user

**Test successful OTP verifications**
* Wait for the OTP iframe to open
* In ~5 minutes, check the Tracks Live View and see that a `platform_checkout_otp_prompt_start` event was recorded with your site's blog ID.
* Enter the correct OTP on the verification prompt
* In ~5 minutes, check the Tracks Live View and see that a `platform_checkout_otp_prompt_complete` event was recorded with your site's blog ID.

**Test failed OTP verifications**
* Repeat the previous steps to get to the OTP prompt 
* Enter an incorrect OTP code
*  In ~5 minutes, check the Tracks Live View and see that a `platform_checkout_otp_prompt_failed` event was recorded with your site's blog ID.

-------------------


**Post merge**




cc @brianyu28
